vevestaX/vevesta.py

```diff
@@ -70,114 +72,160 @@ def featureEngineering(self, value):
-        
+            modeling = pandas.DataFrame(data={**{'experimentID': experimentID,
+                                                 'features': ','.join(self.dataSourcing) + ',' + ','.join(
+                                                     self.featureEngineering),
+                                                 'timestamp in UTC': datetime.utcnow().isoformat()},
+                                              **{k: [v] for k, v in self.variables.items()}}, index=[0])
+
         modeling = pandas.concat([modelingData, modeling], ignore_index=True)
-                
-        #message table       
+
+        # message table
         data = {
-        'experimentID': experimentID,
-        'techniqueUsed': techniqueUsed,
-        'message': message,
-        'version': version,
-        'filename' : self.get_filename(),
-        'timestamp in UTC' : datetime.utcnow().isoformat()
+            'experimentID': experimentID,
+            'techniqueUsed': techniqueUsed,
+            'message': message,
+            'version': version,
+            'filename': self.get_filename(),
+            'timestamp in UTC': datetime.utcnow().isoformat()
         }
-        
-          
-        df_messages = pandas.DataFrame(index =[1], data = data)
+
+        df_messages = pandas.DataFrame(index=[1], data=data)
         df_messages = pandas.concat([messageData, df_messages], ignore_index=True)
- 
-        with pandas.ExcelWriter(filename, engine='openpyxl') as writer: 
-            
-            df_dataSourcing.to_excel(writer, sheet_name = 'dataSourcing', index =False) 
-            
-            df_featureEngineering.to_excel(writer, sheet_name = 'featureEngineering', index = False)
-            modeling.to_excel(writer, sheet_name = 'modelling', index = False)
-        
-            df_messages.to_excel(writer, sheet_name = 'messages', index = False)     
+
+        with pandas.ExcelWriter(filename, engine='openpyxl') as writer:
+
+            df_dataSourcing.to_excel(writer, sheet_name='dataSourcing', index=False)
+
+            df_featureEngineering.to_excel(writer, sheet_name='featureEngineering', index=False)
+            modeling.to_excel(writer, sheet_name='modelling', index=False)
+
+            df_messages.to_excel(writer, sheet_name='messages', index=False)
+
+        if showMessage:
+            message = self.getMessage()
+            print(message)
+
+        return {"messages": data, "df_ds": df_dataSourcing, "df_fe": df_featureEngineering}
+
+    def commit(self, techniqueUsed, filename=None, message=None, version=None, project_id=None):
+        data = self.dump(techniqueUsed, filename=filename, message=message, version=version, showMessage=False)
+
+        messages = data['messages']
+        data_sourced = list(json.loads(data['df_ds'].iloc[-1:].to_json(orient='index')).values())[0]
```

Comment:
Anuran, as discussed use self.data sourcing, self.feature engineering to source values.